### PR TITLE
Sprint 11 Fixes

### PR DESCRIPTION
**Pull request by afrasier**
- Refactor position bid logic into position model (and clean it up a bit)
- Resolve circular dependency
- Change error message

### CODE REVIEW

File: talentmap_api/bidding/models.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 from djchoices import DjangoChoices, ChoiceItem
 
-from talentmap_api.position.models import PositionBidStatistics
+import talentmap_api.position.models
```
Comment (jseppi) [inline talentmap_api/bidding/models.py]: what was this change for?

File: talentmap_api/bidding/views/bidlist.py [hunk 2]
```diff
@@ -111,12 +111,8 @@ def put(self, request, pk, format=None):
...
 
         # Position must not already have a handshake or greater bid status
-        bid_statuses = list(position.bids.filter(bidcycle=bidcycle).values_list("status", flat=True))
-        if (Bid.Status.handshake_offered in bid_statuses or
-                Bid.Status.handshake_accepted in bid_statuses or
-                Bid.Status.in_panel in bid_statuses or
-                Bid.Status.approved in bid_statuses):
-            return Response("Cannot bid on a position with a handshake", status=status.HTTP_400_BAD_REQUEST)
+        if not position.can_accept_new_bids(bidcycle):
+            return Response("Cannot bid on a position that already has a qualifying bid", status=status.HTTP_400_BAD_REQUEST)
```
Comment (jseppi) [inline talentmap_api/bidding/views/bidlist.py]: 👍

File: talentmap_api/bidding/views/bidlist.py [hunk 3]
```diff
@@ -111,12 +111,8 @@ def put(self, request, pk, format=None):
         position = get_object_or_404(bidcycle.positions.all(), id=pk)
 
         # Position must not already have a handshake or greater bid status
-        bid_statuses = list(position.bids.filter(bidcycle=bidcycle).values_list("status", flat=True))
-        if (Bid.Status.handshake_offered in bid_statuses or
-                Bid.Status.handshake_accepted in bid_statuses or
-                Bid.Status.in_panel in bid_statuses or
-                Bid.Status.approved in bid_statuses):
-            return Response("Cannot bid on a position with a handshake", status=status.HTTP_400_BAD_REQUEST)
+        if not position.can_accept_new_bids(bidcycle):
```
Comment (jseppi) [inline talentmap_api/bidding/views/bidlist.py]: 👍

File: talentmap_api/bidding/models.py (hunk 1 above)
Comment (afrasier) [reply talentmap_api/bidding/models.py]: To resolve a circular import issue in the position model file.

Comment (jseppi) [reply talentmap_api/bidding/models.py]: ah, right on

**Pull request merged**